functions/payment.js

```diff
@@ -10,11 +11,60 @@ const STRIPE_CONFIG = functions.config().stripe;
 const stripe = require("stripe")(STRIPE_CONFIG.secret_key);
 
 const PURCHASE_TABLE = "purchases";
+const CUSTOMER_TABLE = "customers";
+
+exports.createCustomerWithPaymentMethod = functions.https.onCall(
+  async (data, context) => {
+    if (!context.auth || !context.auth.token) {
+      throw new functions.https.HttpsError("unauthenticated", "Please log in");
+    }
+
+    if (context.auth.token.aud !== PROJECT_ID) {
+      throw new functions.https.HttpsError(
+        "permission-denied",
+        "Token invalid"
+      );
+    }
+
+    if (!(data.userEmail && data.paymentMethodId && data.userId)) {
+      throw new functions.https.HttpsError(
+        "invalid-argument",
+        "venue, event or return url data missing"
+      );
+    }
+
+    const customer = stripe.customers.create(
+      {
+        email: data.userEmail,
+      },
+      (err, customer) => {
+        if (err) {
+          throw new functions.https.HttpsError("unavailable", err.message);
+        }
+        stripe.paymentMethods.attach(
+          data.paymentMethodId,
+          { customer: customer.id },
+          async (err) => {
+            if (err) {
+              throw new functions.https.HttpsError("unavailable", err.message);
+            }
+            await admin
+              .firestore()
+              .collection(CUSTOMER_TABLE)
+              .doc(data.userId)
+              .set({
+                userId: data.userId,
+                custormerId: customer.id,
+              });
```

Comment:
does it make sense store the stripe id returned in the result of attach in firebase?